modules/spo-replicating-controllers.adoc

```diff
@@ -0,0 +1,131 @@
+// Module included in the following assemblies:
+//
+// * security/security_profiles_operator/spo-seccomp.adoc
+// * security/security_profiles_operator/spo-selinux.adoc
+
+ifeval::["{context}" == "spo-seccomp"]
+:seccomp:
+:type: seccomp
+:kind: SeccompProfile
+endif::[]
+ifeval::["{context}" == "spo-selinux"]
+:selinux:
+:type: SELinux
+:kind: SelinuxProfile
+endif::[]
+
+:_content-type: PROCEDURE
+[id="spo-replicating-controllers_{context}"]
+= Replicating controllers and SecurityContextConstraints
+
+When deploying {type} policies for replicating controllers, such as deployments or daemonsets, note that the `Pod` objects spawned by the controllers are not running with the identity of the user who creates the workload. Unless a `ServiceAccount` is selected, the pods might revert to using a restricted `SecurityContextConstraints` (SCC) which does not allow use of custom security policies.
+
+.Procedure
+
+. Create the following SCC object to allow SELinux policies to be used in the `nginx-secure` namespace:
++
+[source,yaml]
+----
+apiVersion: security.openshift.io/v1
+kind: SecurityContextConstraints
+metadata:
+  annotations:
+    kubernetes.io/description: A special SCC for running nginx with a custom SELinux policy
+  name: nginx-secure
+allowHostIPC: false
+allowHostNetwork: false
+allowHostPID: false
+allowHostPorts: false
+allowPrivilegeEscalation: true
+allowPrivilegedContainer: false
+allowedCapabilities: null
+defaultAddCapabilities: null
+fsGroup:
+  type: MustRunAs
+priority: null
+readOnlyRootFilesystem: false
+requiredDropCapabilities:
+- KILL
+- MKNOD
+- SETUID
+- SETGID
+runAsUser:
+  type: MustRunAsRange
+seLinuxContext:
+  type: MustRunAs
+  seLinuxOptions:
+    type: test-selinux-recording-nginx-0_nginx-secure.process
+supplementalGroups:
+  type: RunAsAny
+users:
+- system:serviceaccount:nginx-secure:nginx-sa
+volumes:
+- configMap
+- downwardAPI
+- emptyDir
+- persistentVolumeClaim
+- projected
+- secret
+----
+
+. Create the Role object:
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: nginx
+  namespace: nginx-secure
+rules:
+- apiGroups:
+  - security.openshift.io
+  resources:
+  - securitycontextconstraints
+  resourceNames:
+  - nginx-secure
+  verbs:
+  - use
+----
+
+. Create the Deployment object:
++
+[source,yaml]
+----
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-deployment
+spec:
+  selector:
+    matchLabels:
+      app: nginx
+  replicas: 2
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      serviceAccountName: nginx-sa
```

Comment:
Better to add the serviceacount and rolebinding to the previouse code block. Use it directly will make users quite confusing.

Reply:
@xiaojiey Can you clarify this? Do you mean it needs to be in the `spec:` block on line 99?

Reply:
But the deployment must reference the SA explicitly, right?

Reply:
@xiaojiey or did you mean to explicitly list the whole manifests that create the SA and the rolebinding? The upstream doc leaves that part for the user and just says "...and finally a role binding and the SA."

If this is confusing, I think it would be best to fix the upstream docs to be explicit as well.

Reply:
Per https://coreos.slack.com/archives/CHCRR73PF/p1670593369064539, there maybe a product bug or more detailed steps needed. I have tested and it not working as expected(maybe something with the steps, still need double confirm).